From: Stephen Cameron <stephenmcameron@gmail.com>
Date: Fri, 23 Jan 2015 16:43:56 -0600
Subject: hpsa: do not check for msi(x) in interrupt_pending
Git-commit: c2b0acde0608542add2bfc2a73c23b98f20eeec4
Patch-mainline: v4.0-rc1
References: bsc#942743

No need to check whether interrupt pending for MSI(X) and
conversely, no need to check whether MSI(X) interrupts are
being used when checking if interrupts are pending.

Reviewed-by: Scott Teel <scott.teel@pmcs.com>
Signed-off-by: Don Brace <don.brace@pmcs.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
Acked-by: Johannes Thumshirn <jthumshirn@suse.com>
---
 drivers/scsi/hpsa.h |    3 ---
 1 file changed, 3 deletions(-)

--- a/drivers/scsi/hpsa.h
+++ b/drivers/scsi/hpsa.h
@@ -485,9 +485,6 @@ static bool SA5_performant_intr_pending(
 	if (!register_value)
 		return false;
 
-	if (h->msi_vector || h->msix_vector)
-		return true;
-
 	/* Read outbound doorbell to flush */
 	register_value = readl(h->vaddr + SA5_OUTDB_STATUS);
 	return register_value & SA5_OUTDB_STATUS_PERF_BIT;
